platforms/nuttx/src/px4/stm/stm32f4/px4io_serial/px4io_serial.cpp

```diff
@@ -432,10 +434,10 @@ ArchPX4IOSerial::_do_interrupt()
 			if ((length < 1) || (length < PKT_SIZE(*_current_packet))) {
 				perf_count(_pc_badidle);
 
-				/* stop the receive DMA */
-				stm32_dmastop(_rx_dma);
+				/* abort dma on error */
+				_abort_dma();
```

Comment:
Because this is a a speak - listen protocol, the TX has to have been done long before the RX can start and the abort is a tad draconian.

Reply:
Done